src/components/views/DownloadModModal.vue

```diff
@@ -167,112 +138,64 @@ interface DownloadProgress {
             const modsWithUpdates: ThunderstoreCombo[] = await this.$store.dispatch('profile/getCombosWithUpdates');
 
-            this.$store.commit('download/increaseAssignId');
-            const currentAssignId = this.$store.state.download.assignId;
+            const assignId = await this.$store.dispatch(
+                'download/addDownload',
+                modsWithUpdates.map(value => `${value.getMod().getName()} (${value.getVersion().toString()})`)
+            );
 
-            const progressObject = {
-                progress: 0,
-                initialMods: modsWithUpdates.map(value => `${value.getMod().getName()} (${value.getVersion().toString()})`),
-                modName: '',
-                assignId: currentAssignId,
-                failed: false,
-            };
-            this.$store.commit('download/setDownloadObject', progressObject);
-            this.$store.commit('download/pushDownloadObjectToAllVersions', {
-                assignId: currentAssignId,
-                downloadObject: progressObject
-            });
             this.downloadingMod = true;
             ThunderstoreDownloaderProvider.instance.downloadLatestOfAll(modsWithUpdates, this.ignoreCache, (progress: number, modName: string, status: number, err: R2Error | null) => {
-                const assignIndex = this.$store.state.download.allVersions.findIndex(([number, val]: [number, DownloadProgress]) => number === currentAssignId);
-                if (status === StatusEnum.FAILURE) {
-                    if (err !== null) {
+                try {
+                    if (status === StatusEnum.FAILURE) {
                         this.downloadingMod = false;
-                        const existing = this.$store.state.download.allVersions[assignIndex];
-                        existing[1].failed = true;
-                        this.$store.commit('download/updateDownloadObject', {
-                            assignId: assignIndex,
-                            downloadVersion: [currentAssignId, existing[1]]
-                        });
-                        DownloadModModal.addSolutionsToError(err);
-                        this.$store.commit('error/handleError', err);
-                        return;
-                    }
-                } else if (status === StatusEnum.PENDING) {
-                    const obj = {
-                        progress: progress,
-                        modName: modName,
-                        initialMods: modsWithUpdates.map(value => `${value.getMod().getName()} (${value.getVersion().getVersionNumber().toString()})`),
-                        assignId: currentAssignId,
-                        failed: false,
+                        this.$store.commit('download/updateDownload', {assignId, failed: true});
+                        if (err !== null) {
+                            DownloadModModal.addSolutionsToError(err);
+                            this.$store.commit('error/handleError', err);
```

Comment:
Nitpick: as a personal preference, we could throw `err` here as we have the try-catch in place anyway. That way we don't have to duplicate the error handling. Applies to `downloadHandler` as well.